packages/lib/src/artifacts/Contract.ts

```diff
@@ -128,13 +128,18 @@ export function createContract(schema: any): Contract {
 
 export function contractMethodsFromAbi(instance: Contract, constant: ContractMethodMutability = ContractMethodMutability.NotConstant): any[] {
   const isConstant = constant === ContractMethodMutability.Constant;
+  const contractAst = new ContractAST(instance, null, { nodesFilter: ['ContractDefinition'] });
+  const methodsFromAst = contractAst.getMethods();
```

Comment:
In which order does `getMethods` return them, in case you have overridden methods? For instance, if you have the following scenario:

```solidity
contract Base is Initializable {
  function foo() initializer { }
}

contract Child is Base {
  function foo() { }
}
```

Then `contractMethodsFromAbi` should return `foo` **without** the initializable flag. But if `getMethods` return from less to most derived (which I think it may be the case), then `find` will hit the first occurrence, and return that it does have an initializer.